# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: 6709a3bc1a1c
# Full Hash: 6709a3bc1a1c4ac18dce0f6db78efe12b3e1f178
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-15 21:39:46
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 83881815f007 -r 6709a3bc1a1c dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Mon Jul 15 11:01:04 2019 +0000
+++ b/dom/ipc/BrowserParent.h	Mon Jul 15 17:30:26 2019 +0000
@@ -274,8 +274,6 @@
   mozilla::ipc::IPCResult RecvAccessKeyNotHandled(
       const WidgetKeyboardEvent& aEvent);
 
-  mozilla::ipc::IPCResult RecvSetHasBeforeUnload(const bool& aHasBeforeUnload);
-
   mozilla::ipc::IPCResult RecvRegisterProtocolHandler(const nsString& aScheme,
                                                       nsIURI* aHandlerURI,
                                                       const nsString& aTitle,
@@ -716,7 +714,6 @@
   void Deprioritize();
 
   bool GetHasContentOpener();
-  bool GetHasBeforeUnload();
 
   bool StartApzAutoscroll(float aAnchorX, float aAnchorY, nsViewID aScrollId,
                           uint32_t aPresShellId);
@@ -942,10 +939,6 @@
   // at least once.
   bool mHasPresented;
 
-  // True if at least one window hosted in the BrowserChild has added a
-  // beforeunload event listener.
-  bool mHasBeforeUnload;
-
   // True when the remote browser is created and ready to handle input events.
   bool mIsReadyToHandleInputEvents;
 